library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java

```diff
@@ -480,7 +480,11 @@ public void updateStatus(int status) {
     }
 
     private boolean isClientReadyToDownload(DownloadBatch downloadBatch) {
-        return downloadClientReadyChecker.isAllowedToDownload(downloadBatch);
+        return downloadClientReadyChecker.isAllowedToDownload(marshallToDownload(downloadBatch));
+    }
+
+    private Download marshallToDownload(DownloadBatch downloadBatch) {
+        return new Download(downloadBatch.getBatchId(), downloadBatch.getCurrentSize(), downloadBatch.getTotalSize());
```

Comment:
I think a Download can be either one item or a batch (as in 1 item is a batch of just 1 item)